### ISSUE TITLE
Outlets are searched in view controller doing the layout node loading

### ISSUE THREAD

**Issue by eimantas** [schibsted/layout#16]
Version: 0.48
Xcode: 9b5
iOS Sim: 11.0

I am duplicating the approach to loading layouts provided by SampleApp. I have a MainViewController that loads `Main.xml` which provides `UITabBarViewController` with a couple of tabs.

The tabs though contain subclasses of `UIViewController` and `LayoutViewController` and the view that is nested under them contain elements with `outlet` attributes. I would expect the `outlet`s  to be connected to the instances of the tabbed view controllers, however I get an error that the outlet could not be found in `MainViewController`.

Is this intended behaviour or a bug?

**Comment by nicklockwood** [comment]
This looks like a bug, although there are some known limitations of the existing binding system, which are that it can't bind some outlets to the main controller and some to intermediate controllers, it has to bind all properties to the same owner, so is it possible that's the cause?

**Comment by eimantas** [comment]
Sadly I can't tell since I don't know much about bindings.

**Comment by nicklockwood** [comment]
Does your MainViewController have any properties whose names match the outlets of your child view controllers?

**Comment by eimantas** [comment]
No. It just loads `Main.xml`

**Comment by nicklockwood** [comment]
Can you paste in your XML? I'll look into it.

**Comment by eimantas** [comment]
Main.xml

```xml
<UITabBarController>

    <UIViewController
        xml="MyTableViewController.xml" />

    <UIViewController
        tabBarItem.title="Scroll"
        xml="ScrollViewController.xml" />
</UITabBarController>
```

ScrollViewController.xml
```xml
<UINavigationController>
    <ScrollViewController>
        <UIView>
        <UIScrollView
            width="100%"
            height="100%"
            backgroundColor="#fff">
            <UIView
                outlet="imageContainer"
                top="0"
                left="0"
                height="236"
                width="100%"
                backgroundColor="#0f0a">
                <UIImageView
                    width="100%"
                    height="100%"/>
            </UIView>
            <UIView
                outlet="firstRow"
                top="previous.bottom"
                left="0"
                height="75"
                width="100%"
                backgroundColor="#ff0a">
                <UILabel
                    top="8"
                    left="8"
                    text="Title" />
                <UILabel
                    top="previous.bottom + 8"
                    left="8"
                    text="Subtitle" />
            </UIView>
            <UIView
                top="previous.bottom"
                width="100%"
                height="75">
                <UILabel
                top="8"
                left="8"
                text="Title" />
                <UILabel
                top="previous.bottom + 8"
                left="8"
                text="Subtitle" />
            </UIView>
        </UIScrollView>
        </UIView>
    </ScrollViewController>
</UINavigationController>
```

ScrollViewController.swift

```swift
import UIKit

import Layout

class ScrollViewController: LayoutViewController {

    var imageContainer: UIView?
    var firstRow: UIView?

    override func viewDidLoad() {
        super.viewDidLoad()
    }
}

```

**Comment by nicklockwood** [comment]
@eimantas sorry it's taken a while to get back to you.

I tried your sample code and I'm not seeing any errors. I thought perhaps I'd fixed the bug by accident, but I tried with 0.4.8 and I'm still not seeing the issue.

I also tried it on both Xcode 8.33 and Xcode 9b5, and it made no difference. If I delete one of the outlets, such as `imageContainer`, then I get an error saying that it doesn't exist on MainViewController, but no error otherwise.

The only thing I could think might be causing it is that `imageContainer` and `firstRow` are not marked with `@objc` or `@IBOutlet`, and Swift 4 is stricter about implicit Obj-C runtime compliance than 3 was, but if you're using Swift 3.2 that shouldn't matter.

Anyway, if you can update the sample, or provide a complete project that repros the issue I'm happy to investigate further.

**Comment by eimantas** [comment]
Here's the project for this issue: https://github.com/eimantas/LayoutAppSandbox

I've bundled it with Layout 0.4.12  version and now instead of receiving error about missing outlets I see the error about missing `MyTableViewController.xml` file.

**Comment by nicklockwood** [comment]
@eimantas OK this is very weird because I just checked out that project you gave me and it runs fine.

I wonder if the issue has something to do with the location on your hard drive. Maybe the path contains spaces or something that's confusing the live reloading?

Another possibility is you may have some junk in your build folder, and a clean build or deleting the app from the simulator might fix it?

**Comment by eimantas** [comment]
Weird. I deleted ModuleCache and project directories in DerivedData and even reset the content and settings for the simulator. Still same error...

**Comment by eimantas** [comment]
Another thing, if I rearrange the `<UIViewController>` tags in `Main.xml` like this:

```xml
<UITabBarController>

    <UIViewController
        tabBarItem.title="Scroll"
        xml="ScrollViewController.xml"
        />

    <UIViewController
        tabBarItem.title="Table"
        xml="MyTableViewController.xml"
        />
</UITabBarController>
```

The app starts complaining about missing `ScrollViewController.xml` file. 🤔

**Comment by nicklockwood** [comment]
Here it is running on my machine, with the Main XML configured as above, and no other changes:

![2017-08-17 15_31_28](https://user-images.githubusercontent.com/546885/29417516-56b488a8-8361-11e7-95b7-8f1a1bdce117.gif)

**Comment by eimantas** [comment]
I removed the precompiled framework and added a reference to Layout.xcodeproj file and linked against the framework built from source code. Issue is the same. Here is the file/project structure:

Layout.framework: /Users/eimantas/Development/layout/
LayoutApp: /Users/eimantas/Desktop/LayoutApp/

So no spaces, no fancy text, just plain ASCII characters. Here's another thing, I've added some print statements around the troubling lines and here's what I got: https://gist.github.com/eimantas/6bb8a779ec9b8106557b9ec52b272931

The separator denotes the call to `LayoutLoader.findSourceURL`. Notice that after finding `Main.xml` it continues searching in framework's directory and never comes back to the application bundle.
